census/vanilla-tf/trainer/task.py

```diff
@@ -0,0 +1,338 @@
+  dense_tensor = tf.sparse_to_indicator(sparse_tensor, vocab_size)
+  dense_tensor = tf.cast(dense_tensor, tf.int32)
+  return dense_tensor
+
+def read_input_tensor(input_file, skiprows=None):
+  """Concatenate the wide columns to produce a single tensor."""
+  inp, label = read_input_data(input_file, skiprows)
+  in_tensor, label_tensor = generate_input(inp, label)
+  return generate_wide_columns(in_tensor), label_tensor
+
+
+#
+# Function to perform the actual training loop.
+# This function is same for single and distributed.
+#
+def training_steps(session, graph, inputs, labels,
+                   max_steps, train_eval_tensor,
+                   job_name='local', job_id=0):
+  """Run the training steps and calculate accuracy every 10 steps."""
+
+  train_step, eval_step, global_step = graph
+  step = tf.train.global_step(session, global_step)
+
+  (train_inp, train_label), (eval_inp, eval_label) = train_eval_tensor
+
+  while step < max_steps:
+    session.run(
+        train_step,
+        feed_dict={
+            inputs: session.run(train_inp),
+            labels: session.run(train_label)
+            }
+    )
+
+    step = tf.train.global_step(session, global_step)
+
+    if step % 10 == 0:
+      accuracy = evaluate_accuracy(session, eval_step, inputs, labels,
+                                   eval_inp, eval_label)
+      print('[{}/{}]: Step number {} of {} done, Accuracy {:.2f}%'.format(
+          job_name, job_id, step, max_steps, accuracy))
+
+  return train_step
+
+
+#
+# Single and Distributed training functions.
+#
+def training_single(job_dir, max_steps, train_eval_tensor):
+  """Perform single node training."""
+
+  inputs, labels = get_placeholders()
+
+  graph = make_graph(inputs, labels)
+  init = tf.global_variables_initializer()
+
+  session = tf.train.MonitoredTrainingSession(checkpoint_dir=job_dir,
+                                              save_checkpoint_secs=20,
+                                              save_summaries_steps=50)
+  session.run(init)
+
+  training_steps(session, graph, inputs, labels, max_steps, train_eval_tensor)
+
+def training_distributed(job_dir, max_steps, train_eval_tensor):
+  """Perform distributed training."""
+
+  # Parse the TF_CONFIG to create cluster spec and job name
+  # We are doing this manually here to demonstrate
+  # See RunConfig abstraction here https://goo.gl/f4BQMo
+  cluster_spec, job_name, task_index = parse_tf_config()
+
+  # Create and start a server
+  server = tf.train.Server(cluster_spec,
+                           job_name=job_name,
+                           task_index=task_index)
+
+  is_chief = (job_name == 'master')
+
+  if job_name == 'ps':
+    server.join()
+  elif job_name in ['master', 'worker']:
+    with tf.device(tf.train.replica_device_setter()):
+      inputs, labels = get_placeholders()
+      graph = make_graph(inputs, labels)
+
+    init = tf.global_variables_initializer()
+
+    with tf.train.MonitoredTrainingSession(master=server.target,
```

Comment:
The graph needs to declare various summary ops, as well as a tf.summary.merge_all op. That would be used to produce summaries.

Secondly, do you need to explicitly declare a Saver, for the checkpointing to use?